Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs

```diff
@@ -53,32 +56,30 @@ public async Task<ActionResult> Post()
                 _logger.LogError("File given to image posting was null.");
                 return BadRequest();
             }
-            try
-            {
-                // generate a random guid from the file name
-                string extension = file
-                    .FileName
-                        .Split('.')
-                        .Last();
-                string newFileName = $"{Request.Form["userId"]}-{Guid.NewGuid()}.{extension}";
-                _logger.LogInformation($"New file named to be uploaded, {newFileName}");
 
-                var result = await _storageService.UploadFileContentAsync(
-                        file.OpenReadStream(),
-                        ProfileConfiguration.BlobContainerName,
-                        file.ContentType,
-                        newFileName);
+            // generate a random guid from the file name
+            string extension = file
+                .FileName
+                    .Split('.')
+                    .Last();
+            string newFileName = $"{Request.Form["userId"]}-{Guid.NewGuid()}.{extension}";
+            _logger.LogInformation($"New file named to be uploaded, {newFileName}");
 
-                var toReturn = result.AbsoluteUri;
-                _logger.LogInformation($"New file uploaded, {newFileName}");
+            var profileConfigOptions = new ProfileConfigOptions();
+            _configuration.GetSection(ProfileConfigOptions.ProfileConfig).Bind(profileConfigOptions);
```

Comment:
this isn't how we do options pattern. this code goes in Startup.ConfigureServices, and you inject IOptions<>, not the whole configuration. but if it works it works

Reply:
I guess I came across an older version of Options pattern, instead of injecting it with IOptions<>